pkgs/desktops/gnome-3/extensions/default.nix

```diff
@@ -0,0 +1,38 @@
+{ lib
+, callPackage
+}:
+let
+  buildShellExtension = callPackage ./buildGnomeExtension.nix { };
+
+  # Index of all scraped extensions (with supported versions)
+  extensionsIndex = lib.importJSON ./extensions.json;
+
+  # Take all extensions from the index that match the gnome version, build them and put them into an attrset that is ready to go into nixpkgs
+  produceExtensions = shell-version:
+    lib.trivial.pipe extensionsIndex [
+      (builtins.filter
+        # Does a given extension match our current shell version?
+        (extension: (builtins.hasAttr shell-version extension."shell_version_map"))
+      )
+      (map
+        # Take in an `extension` object from the JSON and transform it into the correct args to call `buildShellExtension`
```

Comment:
Shouldn't each `map` block have its comment before the opening `(`?  Would help noticing that these are indeed different elements in the list and not chained map calls.